modules/virt-binding-devices-vfio-driver.adoc

```diff
@@ -0,0 +1,132 @@
+<2> Applies the new kernel argument only to worker nodes.
+<3> The number (100) indicates the rank for this `MachineConfig` among all `MachineConfig` objects. The name given to this `MachineConfig` indicates the purpose of this object.
+<4> The encoded text for `source` (`b3B0aW9ucyB2ZmlvLXBjaSBpZHM9MTBkZToxZWI4Cg`) translates to `options vfio-pci ids=10de:1eb8`. The `vendor-ID` value (10de) and the `device-ID` value (1eb8) applies to a single device that binds to the VFIO driver. You can add a list of multiple devices for `source` with their vendor and device information.
+<5> Description needed. What does the value 420 for `mode` do for this source file? Please provide details.
+<6> Description needed. When `mode` is set to `true`, is there a default file that is overwritten by the `/etc/modprobe.d/vfio.conf` file? If that's not correct, please provide details.
+<7> The path for the `vfio.conf` file.
+<8> The encoded text for `source` (`dmZpby1wY2kK`) that translates to `options vfio-pci" ids=10de:1eb8|base64`.
+<9> The file with the `vfio-pci` configuration that loads the PCI device with the VFIO driver.
+
+. Create a `MachineConfig` object file that identifies the kernel argument for the IOMMU configuration:
++
+[source,terminal]
+----
+$ oc create -f 100-worker-kernel-arg-iommu.yaml
+----
+
+. Verify that the `MachineConfig` object file was added.
++
+[source,terminal]
+----
+$ oc get MachineConfig
+----
++
+.Example output
+[source, terminal]
+----
+NAME                             GENERATEDBYCONTROLLER                      IGNITIONVERSION  AGE
+00-master                        d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+00-worker                        d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+01-master-container-runtime      d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+01-master-kubelet                d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+01-worker-container-runtime      d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+01-worker-kubelet                d3da910bfa9f4b599af4ed7f5ac270d55950a3a1   3.2.0            25h
+100-worker-iommu                                                            3.2.0            30s
+100-worker-vfiopci-configuration                                            3.2.0            30s
+
+----
+
+. Use the `lspci` command to obtain the `vendor-ID` and the `device-ID` for the PCI device.
++
+[source, terminal]
+----
+$ lspci -nnv | grep -i nvidia
+----
+.Example output
+[source, terminal]
+02:01.0 3D controller [0302]: NVIDIA Corporation GV100GL [Tesla V100 PCIe 32GB] [10de:1eb8] (rev a1)
+
+. Edit the VFIO configuration in the `etc/modprobe.d/vfio.conf` file by adding the values for the `vendor-ID` and the `device-ID`.
++
+[source,terminal]
+----
+$ echo "options vfio-pci ids=10de:1eb8 > /etc/modprobe.d/vfio/conf"
+----
+These values are converted into the `base64 format` and encoded.
```

Comment:
the `/etc/modprobe.d/vfio.conf` is the file that will be generated by the MachineConfig Operator after the MachineConfig Object is deployed on one of the worker nodes. 
There is no need to edit the file. Instead, we should say here that the administrator needs to take the values of the vendor:device ids, add it to the `options vfio-pci ids=....` and convert this line to base64 and then add this line to the `source`  <4> - on line 33 of your MachineConfig example

I think instead of the above we should write something like:

. Convert the content of the /etc/modprobe.d/vfio.conf file to base64:
---
$ echo options vfio-pci ids=10de:1eb8 | base64
---
The output will be `b3B0aW9ucyB2ZmlvLXBjaSBpZHM9MTBkZToxZWI4Cg==`

Add the encoded content to the `source` field <4> as in the example below.

Reply:
Updated the content. Please review.